engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala

```diff
@@ -4,36 +4,68 @@ import io.confluent.kafka.serializers._
+import pl.touk.nussknacker.engine.kafka.serialization.{KafkaVersionAwareKeyValueDeserializationSchemaFactory, KafkaVersionAwareValueDeserializationSchemaFactory}
 
 trait ConfluentKafkaAvroDeserializerFactory {
   import collection.JavaConverters._
 
-  protected def createDeserializer[T](schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory, kafkaConfig: KafkaConfig, isKey: Boolean, useSpecificAvroReader: Boolean): Deserializer[T] = {
+  protected def createDeserializer[T](mode: ConfluentKafkaAvroDeserializationMode,
+                                      topic: String,
+                                      version: Option[Int],
+                                      schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory,
+                                      kafkaConfig: KafkaConfig,
+                                      isKey: Boolean,
+                                      useSpecificAvroReader: Boolean): Deserializer[T] = {
     val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
-    val deserializer = new KafkaAvroDeserializer(schemaRegistryClient.client)
+
+    val deserializer = mode match {
+      case ConfluentKafkaAvroDeserializationMode.STATIC =>
+        ConfluentStaticKafkaAvroDeserializer(schemaRegistryClient, topic, version, isKey = isKey)
+      case ConfluentKafkaAvroDeserializationMode.FRESH =>
+        new ConfluentKafkaAvroDeserializer(schemaRegistryClient, topic, version, isKey = isKey)
+      case _ =>
+        new KafkaAvroDeserializer(schemaRegistryClient.client)
+    }
+
     val props = kafkaConfig.kafkaProperties.getOrElse(Map.empty) + (
       KafkaAvroDeserializerConfig.SPECIFIC_AVRO_READER_CONFIG -> useSpecificAvroReader
     )
+
     deserializer.configure(props.asJava, isKey)
     deserializer.asInstanceOf[Deserializer[T]]
   }
 }
 
-class ConfluentKafkaAvroDeserializationSchemaFactory[T: TypeInformation](schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory, useSpecificAvroReader: Boolean)
+class ConfluentKafkaAvroDeserializationSchemaFactory[T: TypeInformation](mode: ConfluentKafkaAvroDeserializationMode,
+                                                                         schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory,
+                                                                         useSpecificAvroReader: Boolean)
   extends KafkaVersionAwareValueDeserializationSchemaFactory[T] with ConfluentKafkaAvroDeserializerFactory {
 
   override protected def createValueDeserializer(topics: List[String], version: Option[Int], kafkaConfig: KafkaConfig): Deserializer[T] =
-    createDeserializer[T](schemaRegistryClientFactory, kafkaConfig, isKey = false, useSpecificAvroReader)
+    createDeserializer[T](mode, topics.head, version, schemaRegistryClientFactory, kafkaConfig, isKey = false, useSpecificAvroReader)
```

Comment:
Please throw an exception when topic list is > 1

Reply:
Mayby this error should be rised in upper place? Why we put there list of topics? I can say - it's not   part of my change, but why now speak about it? :)

Reply:
Wy have here list of topics because we can configure list of topics in KafkaSource. It shouldn't be rised on upper level because on this level of code this constraint need to be fulfilled.
> why we now speak about it
I don't get it. It is ok, that you have list of topics here. I only want from you to add assertion in case if someone will use `ConfluentKafkaAvroDeserializationSchemaFactory` with multi-topic source that we don't support

Reply:
See https://en.wikipedia.org/wiki/Defensive_programming

Reply:
Done ;)